pkg/controller/operconfig/migration.go

```diff
@@ -408,18 +410,41 @@ func convertOvnEgressIpToSdnEgressIp(ctx context.Context, client cnoclient.Clien
 				}
 				egressCIDRsArr = ovnMigrationNodeAnnotation.EgressCIDRs
 			} else {
-				egressIpConfig := node.Annotations[egressIPNodeConfig]
-				egressIpConfigArr := make([]NodeEgressIpConfig, 0)
-				if err := json.Unmarshal([]byte(egressIpConfig), &egressIpConfigArr); err != nil {
-					return err
-				}
-				if len(egressIpConfigArr) <= 0 {
-					return fmt.Errorf("unexpected error: egress-ipconfig annotation is empty")
-				}
-				nodeSubnet, ok := egressIpConfigArr[0].IfAddr["ipv4"]
-				if !ok {
-					return fmt.Errorf("unexpected error: egress-ipconfig annotation missing ipv4 entry")
+				egressIpConfig, ok := node.Annotations[egressIPNodeConfig]
+				var nodeSubnet string
+				if ok {
+					egressIpConfigArr := make([]NodeEgressIpConfig, 0)
+					if err := json.Unmarshal([]byte(egressIpConfig), &egressIpConfigArr); err != nil {
+						return err
+					}
+					if len(egressIpConfigArr) <= 0 {
+						return fmt.Errorf("unexpected error: egress-ipconfig annotation is empty")
+					}
+					nodeSubnet, ok = egressIpConfigArr[0].IfAddr["ipv4"]
+					if !ok {
+						return fmt.Errorf("unexpected error: egress-ipconfig annotation missing ipv4 entry")
+					}
+				} else {
+					// on non-cloud platforms there's no egressIpConfig annotation, use node's primary ifaddr annotation
+					nodeIfAddrAnno, ok := node.Annotations[ovnAnnotationNodeIfAddr]
+					if !ok {
+						return fmt.Errorf("unexpected error: node annotation missing node subnet entry")
+					}
+					nodeIfAddr := make(map[string]string)
+					if err := json.Unmarshal([]byte(nodeIfAddrAnno), &nodeIfAddr); err != nil {
+						return err
+					}
+					address, ok := nodeIfAddr["ipv4"]
+					if !ok {
+						return fmt.Errorf("unexpected error: node annotation missing ipv4 entry")
+					}
+					_, cidr, err := net.ParseCIDR(address)
```

Comment:
Why are you parsing the CIDR here but not doing that for the cloud case?

Reply:
In cloud case, we can use node annotation `cloud.network.openshift.io/egress-ipconfig` to derive the node underlay subnet, which is not available in non-cloud platforms.